SmartPlaces.Facilities/samples/Telemetry.Mapped/src/Processors/TelemetryIngestionProcessor.cs

```diff
@@ -0,0 +1,542 @@
+        where TOptions : IngestionManagerOptions
+    {
+        private readonly DigitalTwinsClient digitalTwinsClient;
+        private readonly ILogger<TelemetryIngestionProcessor<TOptions>> logger;
+        private readonly TelemetryClient telemetryClient;
+        private readonly ITwinMappingIndexer twinMappingIndexer;
+        private readonly TOptions options;
+
+        // Target Telemetry Values
+        private const string telemetryValueRoot = "lastKnownValue";
+        private const string telemetryValueKey = "value";
+        private const string telemetryTimestampKey = "timestamp";
+
+        private static IDictionary<string, DTEntityKind> modelIdToTypeCache = new ConcurrentDictionary<string, DTEntityKind>();
+
+        //Metrics
+        private static readonly string metricsNamespace = "Telemetry";
+        private readonly MetricIdentifier twinUpdateMetric = new MetricIdentifier(metricsNamespace, "TwinUpdate","Status","Reason");
+        private readonly MetricIdentifier telemetryTypeMetric = new MetricIdentifier(metricsNamespace, "TelemetryType", "SourceType", "TargetType");
+
+        /// <summary>
+        /// Setup for how to process an event flowing across an EventHub
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="options"></param>
+        /// <param name="twinMappingIndexer"></param>
+        /// <param name="httpClientFactory"></param>
+        /// <param name="telemetryClient"></param>
+        public TelemetryIngestionProcessor(ILogger<TelemetryIngestionProcessor<TOptions>> logger,
+                                           IOptions<TOptions> options,
+                                           ITwinMappingIndexer twinMappingIndexer,
+                                           IHttpClientFactory httpClientFactory,
+                                           TelemetryClient telemetryClient)
+        {
+            this.logger = logger;
+            this.options = options.Value;
+            this.twinMappingIndexer = twinMappingIndexer;
+            this.telemetryClient = telemetryClient;
+
+            var httpClient = httpClientFactory.CreateClient("Microsoft.SmartPlaces.Facilities");
+
+            logger.LogInformation("TwinMappingIndexer: {type}", twinMappingIndexer.GetType().Name);
+            digitalTwinsClient = new DigitalTwinsClient(new Uri(this.options.AzureDigitalTwinsEndpoint), new DefaultAzureCredential(), new DigitalTwinsClientOptions
+            {
+                Transport = new HttpClientTransport(httpClient),
+            });
+        }
+
+        /// <summary>
+        /// This method defines what actions should be taken to translate from source telemetry data and ingest
+        /// into Azure Digital Twins.
+        /// </summary>
+        /// <param name="telemetryEvent">Contains the values passed through the IotHub into the EventHub to be procedded by Telemetry</param>
```

Comment:
procedded by Telemetry?

Reply:
Oh wow.. VisualStudio failed me. Spelling mistakes all over the place.